src/tests/test_magic.py

```diff
@@ -187,6 +187,23 @@ def test_persist_no_index(ip):
     assert persisted == [(1, "foo"), (2, "bar")]
 
 
+@pytest.mark.parametrize(
+    "sql_statement, expected_error",
+    [
+        ("%%sql --stuff\n SELECT * FROM test", "Unrecognized argument(s)"),
```

Comment:
will this work if the token is quoted? we need to add a test

e.g., this should not fail:

```
%%sql --table '--some' 
```

it might throw a different error (I'm unsure), but it should not be a usage error because --table is valid and '--some' is an argument passed to `--table`, not an argument itself

Reply:
I test it. it throws an AssertionError.  `assert (UsageError('Expected "\'--some\'\\n SELECT * FROM test" to be a pd.DataFrame but it\'s not a valid identifier`
So, we don't want this test fail?

Reply:
that's fine. we don't want the error to be a "Unrecognized argument", if it's failing because the identifier is incorrect (as you're showing), that's fine. just add a new test and validate that the error thrown is not  a "Unrecognized argument"

Reply:
fix it now